Sources/HTTPParser/http_parser.swift

```diff
@@ -61,20 +173,44 @@ private let HTTP_MAX_HEADER_SIZE = 80*1024
  * many times for each string. E.G. you might get 10 callbacks for "on_url"
  * each providing just a few characters more data.
  */
-public protocol http_parser_delegate {
-  func on_message_begin() -> Int
-  func on_url(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_status(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_header_field(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_header_value(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_headers_complete() -> Int
-  func on_body(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_message_complete() -> Int
+public struct http_parser_delegate {
```

Comment:
Here is a performance comparison with the two scenarios I mentioned above.  I cannot explain why protocol http_parser_delegate + struct http_parser is the slowest combination, but it seems like extra retain/release calls are involved.
<img width="1296" alt="protocol" src="https://cloud.githubusercontent.com/assets/7662738/21169872/ff74d4b6-c174-11e6-85c4-e88b6277db01.png">